### PR TITLE
doc: Explain that feedback needs to be addressed

### PR DESCRIPTION
**Pull request by maflcko**
Generally, the pull request author is expected to reply to all comments or iterate the code before merge. Of course, it is allowed to reject feedback, but it should not be done by silently ignoring it.

Clarify this in the docs.

Also, some minor copy edits.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (brunoerg) [inline CONTRIBUTING.md:194]: "As the pull request author you are expected to address any review comments
before your pull request is merged" - What do you mean by 'address'?

Comment (prayank23) [reply CONTRIBUTING.md:194]: If @MarcoFalke responds to this comment and explains the meaning of 'address' in this context, it will be 'addressing' review comments of this pull request.

In other pull requests it can be related to code and sometimes reviewers may not agree with certain changes or rationale provided for the code.

Comment (brunoerg) [reply CONTRIBUTING.md:194]: Make sense. I thought it could sound 'implement' what it is not true since we may not agree with all feedbacks. But it sounds like 'reply' or 'consider' what makes sense. Thank you for explaining, just a doubt related to this meaning.

Comment (MarcoFalke) [reply CONTRIBUTING.md]: Thanks, changed a bit.

File: CONTRIBUTING.md
Comment (michaelfolkson) [inline CONTRIBUTING.md:195]: I think "interate" is a typo. What is this supposed to be, "iterate"? I think "update" or "adjust" would be better here?

File: CONTRIBUTING.md
Comment (michaelfolkson) [inline CONTRIBUTING.md:194]: Here it is "as the pull request author **you** are". But later it moves from "you" back to the PR "author may need to..."

Try to be consistent? Either it is all "you should" or "the PR author should". I think I prefer the latter here. It gets confusing if it is not clear whether "you" is the PR author or the PR reviewer.

**Review by michaelfolkson — COMMENTED** [review]
Concept ACK, Approach ACK

Generally these changes are improvements, just a couple of nits.

**Review by shaavan — APPROVED** [review]
ACK fad23430870f6acf3b343302a02aae97131db689

Going commit wise:
- Adding a link to “release cycle” documentation and mentioning it at the beginning of documentation removes the need for a separate Release Policy section. Also, a reader would better understand a release cycle and the role of a “lead maintainer” in it.
- I agree that the note in the `Address Feedback` section better belongs to the Peer Review section.
- Removing “until you have satisfied all feedback.” increases the sentence’s brevity. Also, the added paragraph makes the reader understand the need to address the feedback on their PRs. This is an essential point and habit that every new contributor should follow.

File: CONTRIBUTING.md
Comment (MarcoFalke) [reply CONTRIBUTING.md:195]: Thanks, fixed

File: CONTRIBUTING.md
Comment (MarcoFalke) [reply CONTRIBUTING.md]: This section uses "you", the others do not, so I stuck to that.

**Comment by michaelfolkson** [comment]
ACK fa694f61ab21cb973843e5234a6aeacd4a957e05

File: CONTRIBUTING.md
Comment (prayank23) [inline CONTRIBUTING.md:197]: Last sentence can be interpreted in different ways and used to close pull requests. I am not sure about this. Do you have any examples or can this be rephrased to make things clear?

Comment (MarcoFalke) [reply CONTRIBUTING.md:197]: Example: https://github.com/bitcoin/bitcoin/pull/13360#issuecomment-673306144

**Review by ghost — APPROVED** [review]
ACK https://github.com/bitcoin/bitcoin/pull/24433/commits/fa694f61ab21cb973843e5234a6aeacd4a957e05

This PR improves docs. I was not sure about one line but example provided by PR author looks okay.

**Comment by Sjors** [comment]
ACK a694f61ab21cb973843e5234a6aeacd4a957e05

**Comment by jamesob** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/24433/commits/fa694f61ab21cb973843e5234a6aeacd4a957e05

**Review by brunoerg — APPROVED** [review]
ACK fa694f61ab21cb973843e5234a6aeacd4a957e05

**Comment by Pragmatic2021** [comment]
I'm trying to open this the wrong way. Now I know. Since you reached out to me I'll try to get back quickly. Thank you. I think whoever comes across anything amiss, you have my permission to make any changes you feel necessary. I'm still kinda finding my way around and I do thank you so much. I'll try to keep up if you need me to approve anything. I trust your judgement implicitly. 
Catherine

**Comment by Pragmatic2021** [comment]
I'm sorry, I finally found the comment button. I looked at everything and from what I can tell you I'm happy with my repository so far. Where does it go now it's finished? I'm assuming it's finished, either way ,I have no complaints.

**Review by w0xlt — APPROVED** [review]
ACK fa694f6

**Pull request merged**
